src/DynamoCoreWpf/ViewModels/PackageManager/PackageViewModel.cs

```diff
@@ -194,7 +194,16 @@ private void Deprecate()
 
         private bool CanDeprecate()
         {
-            return dynamoViewModel.Model.AuthenticationManager.HasAuthProvider;
+            PackageInfo pkg = new PackageInfo(Model.Name, new Version(Model.VersionName));
+            bool isAuthor = false;
+            var header = packageManagerClient.GetPackageHeader(pkg);
+            if (header != null) {
+                if (header.maintainers.First().username.Equals(dynamoViewModel.Model.AuthenticationManager.Username))
+                {
+                    isAuthor = true;
+                }
+            }
+            return dynamoViewModel.Model.AuthenticationManager.HasAuthProvider && isAuthor;
```

Comment:
The `isAuthor` check involves a call to the server through `GetPackageHeader`.  The `HasAuthProvider` check is cheap, and doesn't require a call to the server.  It would be better to do that check up front, and only do the `GetPackageHeader` call if it's true.

Reply:
if !dynamoViewModel.Model.AuthenticationManager.HasAuthProvide {
        return false
    }

    PackageInfo pkg = new PackageInfo(Model.Name, new Version(Model.VersionName));
    var header = packageManagerClient.GetPackageHeader(pkg);
    return ((header != null) && (header.maintainers.Any(...)));

Reply:
agreed!